Java/src/main/java/com/gildedrose/GildedRose.java

```diff
@@ -9,54 +9,66 @@ public GildedRose(Item[] items) {
 
     public void updateQuality() {
```

Comment:
good naming of conditions but still a lot of complex if-else logic in one class (one class needs to know all possible rules like this)